pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -0,0 +1,107 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	ScrapeConfigKind    = "ScrapeConfig"
+	ScrapeConfigName    = "scrapeconfigs"
+	ScrapeConfigKindKey = "scrapeconfig"
+)
+
+// +genclient
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:categories="prometheus-operator",shortName="scfg"
+// +kubebuilder:storageversion
+
+// ScrapeConfig defines a namespace ScrapeConfig to be aggregated across multiple namespaces inta  prometheus
+// configuration
+type ScrapeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ScrapeConfigSpec `json:"spec"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+func (l *ScrapeConfig) DeepCopyObject() runtime.Object {
+	return l.DeepCopy()
+}
+
+// ScrapeConfigList is a list of ScrapeConfigs.
+// +k8s:openapi-gen=true
+type ScrapeConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of ScrapeConfigs
+	Items []*ScrapeConfig `json:"items"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+// TODO: understand what this does
+func (l *ScrapeConfigList) DeepCopyObject() runtime.Object {
+	return l.DeepCopy()
+}
+
+// ScrapeConfigSpec is a specification of the desired configuration for a scrape configuration.
+type ScrapeConfigSpec struct {
+	StaticConfigs []StaticConfig     `json:"staticConfigs,omitempty"`
+	FileSDConfigs []FileSDConfig     `json:"fileSDConfigs,omitempty"`
+	HTTPSDConfigs []HTTPSDConfig     `json:"httpSDConfigs,omitempty"`
+	Relabelings   []v1.RelabelConfig `json:"relabelings,omitempty"`
+	MetricsPath   string             `json:"metricsPath,omitempty"`
+}
+
+// StaticConfig defines a prometheus static configuration
+// See TODO: add link
+type StaticConfig struct {
+	// List of targets for this scrape configuration
+	Targets []string `json:"targets"`
+	// List of labels to add to the metrics scrape as part of this configuration
+	// +optional
+	Labels []string `json:"labels,omitempty"` // TODO: move to a proper Label definition
+}
+
+// FileSDConfig defines a prometheus file service discovery configuration
+// See TODO: add link
+type FileSDConfig struct {
+	// List of files to be used for file discovery. The paths are relative to the current working directory of the
+	// prometheus container. Files must be mounted using Prometheus.ConfigMaps.
```

Comment:
configmaps or secrets?

Reply:
Both?

Reply:
Like the EnvValueFrom of the Pod:
https://cs.opensource.google/go/x/build/+/ce55604c:kubernetes/api/types.go;l=799

Reply:
Indeed, both should work, fixed in 709fa1c13